### PR TITLE
feat: PHPUnit 10.3 support

### PR DESCRIPTION
**Pull request by michalbundyra**

### CODE REVIEW

File: classes/DefaultArgumentRemoverReturnTypes100.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function matches(Invocation $invocation) : bool
                 $iClass ? Invocation::class : Invocation\StaticInvocation::class
             );
         } else {
-            MockFunctionGenerator::removeDefaultArguments($invocation->parameters);
+            MockFunctionGenerator::removeDefaultArguments($invocation->parameters());
```
Comment (marcoscoelho) [inline classes/DefaultArgumentRemoverReturnTypes100.php:41]: fix only vars can be passed as reference.

```suggestion
            $paramaters = $invocation->parameters();
            MockFunctionGenerator::removeDefaultArguments($paramaters);
```

File: classes/DefaultArgumentRemoverReturnTypes100.php [hunk 2]
```diff
@@ -73,7 +73,7 @@ public function toString() : string
     private function removeDefaultArguments(Invocation $invocation, string $class)
     {
         $remover = function () {
-            MockFunctionGenerator::removeDefaultArguments($this->parameters);
+            MockFunctionGenerator::removeDefaultArguments($this->parameters());
```
Comment (marcoscoelho) [inline classes/DefaultArgumentRemoverReturnTypes100.php:76]: fix only vars can be passed as reference.

```suggestion
            $paramaters = $this->parameters();
            MockFunctionGenerator::removeDefaultArguments($paramaters);
```

**Review by marcoscoelho — CHANGES_REQUESTED** [review]
however, if the var cannot be changed by ref. it will not make sense to call the implementation.

**Comment by michalbundyra** [comment]
Closing in favour of #63

**Pull request closed without merging**